pilot/pkg/xds/filters/filters.go

```diff
@@ -104,7 +105,9 @@ var (
 	OriginalSrc = &listener.ListenerFilter{
 		Name: OriginalSrcFilterName,
 		ConfigType: &listener.ListenerFilter_TypedConfig{
-			TypedConfig: util.MessageToAny(&originalsrc.OriginalSrc{}),
+			TypedConfig: util.MessageToAny(&originalsrc.OriginalSrc{
+				Mark: uint32(os.Getegid()),
```

Comment:
I don't understand why this is used as the mark?!

Reply:
Should this be set as `1337`?

Reply:
pod template provided by  configmap `istio-sidecar-injector` declares `runAsGroup: 1337` for TPROXY mode:
```
          privileged: {{ .Values.global.proxy.privileged }}
          readOnlyRootFilesystem: {{ not .Values.global.proxy.enableCoreDump }}
          runAsGroup: 1337
          fsGroup: 1337
          {{ if or (eq (annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode) `TPROXY`) (eq (annotation .ObjectMeta `sidecar.istio.io/capNetBindService` .Values.global.proxy.capNetBindService) `true`) -}}
          runAsNonRoot: false
          runAsUser: 0
          {{- else -}}
          runAsNonRoot: true
          runAsUser: 1337
          {{- end }}
```
I think it's more portable by get `1337` dynamically ?

Reply:
I did not realize that the istiod is run as 1337, @howardjohn I remember you said  there is a case we run istiod locally without in k8s.

Reply:
we should not assume istiod uid/gid correlates. we set it to 1337 only out of convenience

Reply:
my misreading... should we introduce an environment variable or just hardcode 1337?